### PR TITLE
Update db schema

### PR DESCRIPTION
**Pull request by AlexanderGeiger**
Hi,

I implemented the changes to the db schema mentioned [here](https://github.com/D3-AI/Orion/pull/57) & [here](https://github.com/D3-AI/Orion/issues/48).
I also modified the Orion Explorer to work with the new structure. 
The proposed interaction with the Orion Explorer would be:
```
explorer = OrionExplorer(database='orion')
# 1. Add a pipeline
explorer.add_pipeline('lstm_dynamic_threshold', '../orion/pipelines/lstm_dynamic_threshold.json')
# 2. Add a dataset
explorer.add_dataset('testdataset')
# 3. Add signals to dataset
explorer.add_signal('E-1', 'testdataset')
explorer.add_signal('S-1', 'testdataset')
# 4. Run experiment
explorer.run_experiment(name='testexperiment', project='testproject', pipeline='lstm_dynamic_threshold', dataset='testdataset')
```
`run_experiment()` will execute a Datarun for each Signal in the Dataset. 

@csala @dyuliu Can you please have a look if I am missing something? 
Also, there are multiple ways to implement this workflow. So if you have any suggestions how to improve the whole Orion Explorer interaction, please let me know. Thanks.

### CODE REVIEW

File: orion/analysis.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import logging
 
 import pandas as pd
+
```
Comment (csala) [inline orion/analysis.py]: This blank line should not be here. This is the cause for the CI error.

File: orion/explorer.py [hunk 2]
```diff
@@ -5,10 +5,10 @@
 import pandas as pd
 from bson import ObjectId
 from bson.errors import InvalidId
-from mlblocks import MLPipeline
 from mongoengine import connect
 from pip._internal.operations import freeze
 
+from mlblocks import MLPipeline
```
Comment (csala) [inline orion/explorer.py]: This line should be where it was before.

File: orion/explorer.py [hunk 3]
```diff
@@ -114,30 +131,79 @@ def get_pipelines(self, name=None):
...
+                raise ValueError('Pipeline not found: {}'.format(pipeline))
+            else:
+                return found_pipeline
 
     def load_pipeline(self, pipeline):
         LOGGER.info("Loading pipeline %s", pipeline.name)
         return MLPipeline.from_dict(pipeline.mlpipeline)
 
-    def get_dataruns(self, dataset=None, pipeline=None):
+    def run_experiment(self, name, project, pipeline, dataset, user_id=None):
```
Comment (csala) [inline orion/explorer.py]: Can you make this the last method on the class (just move it down)?
The rationale behind it is that it uses other methods from the class which are defined further down, and all the code is sorted following the "always try to define methods before the other methods that use them" principle.

File: orion/model.py [hunk 4]
```diff
@@ -95,8 +95,12 @@ def validate(self, value):
 
 class Dataset(Document, MongoUtils):
     name = fields.StringField(required=True)
-    signal_set = fields.StringField(required=True)
-    satellite_id = fields.StringField()
+    entity_id = fields.StringField()
```
Comment (csala) [inline orion/model.py]: Why is this called `entity_id` instead of `satellite_id`?

File: orion/model.py [hunk 5]
```diff
@@ -111,9 +115,17 @@ class Pipeline(Document, MongoUtils):
     created_by = fields.StringField()
 
 
-class Datarun(Document, MongoUtils):
-    dataset = fields.ReferenceField(Dataset)
+class Experiment(Document, MongoUtils):
+    name = fields.StringField(required=True)
```
Comment (csala) [inline orion/model.py]: Do we need a name for the experiment?
I think we could skip this field and just have the internal `_id` and the `insert_time`.

File: orion/model.py (hunk 4 above)
Comment (AlexanderGeiger) [reply orion/model.py]: I think `entity_id` is a more general term, as we could also have other datasets that do not come from satellites in the future.

**Comment by dyuliu** [comment]
> In general it LGTM, but I added a couple of comments about styling and a question about whether we need to have an Experiment.name or not.
> 
> Once the styling issues and the question are resolved, it looks good to merge!

This logic is good!

**Pull request merged**